jeongjaein/ElevenStreetSample/app/src/main/java/com/exam/elevenstreet/view/product/MainActivity.kt

```diff
@@ -0,0 +1,46 @@
+package com.exam.elevenstreet.view.product
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.exam.elevenstreet.R
+import com.exam.elevenstreet.view.product.adapter.ProductRecyclerViewAdapter
+import com.exam.elevenstreet.view.product.presenter.MainContract
+import com.exam.elevenstreet.view.product.presenter.ProductPresenter
+import com.example.elevenstreet.ProductResponse
+import kotlinx.android.synthetic.main.activity_main.*
+
+
+class MainActivity : AppCompatActivity(), MainContract.View {
+    override fun showProduct(productList: List<ProductResponse>) {
+
+    }
+
+
+    override var presenter: MainContract.Presenter = ProductPresenter()
+
+    private val productRecyclerViewAdapter by lazy { ProductRecyclerViewAdapter() }// 컴파일때 생성말고 접근할때 생성
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
```

Comment:
```suggestion
presenter = ProductPresenter()
```
Presenter에 repository, view를 parameter로 넣어야하기에 lateinit을 사용해야 할 것 같습니다.

Reply:
반영하겠습니당